### PR TITLE
*: show and information_schema should be case insensitive 

### LINKED ISSUE #29910: `show full columns like` should be case insensitive
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
mysql> create table t(id int);
Query OK, 0 rows affected (0.01 sec)

mysql> show full columns from t like 'id';
+-------+---------+-----------+------+------+---------+-------+---------------------------------+---------+
| Field | Type    | Collation | Null | Key  | Default | Extra | Privileges                      | Comment |
+-------+---------+-----------+------+------+---------+-------+---------------------------------+---------+
| id    | int(11) | NULL      | YES  |      | NULL    |       | select,insert,update,references |         |
+-------+---------+-----------+------+------+---------+-------+---------------------------------+---------+
1 row in set (0.00 sec)

mysql> show full columns from t like 'ID';
Empty set (0.00 sec)
```

### 2. What did you expect to see? (Required)

In MySQL 8.0:
```sql
mysql> show full columns from t like 'ID';
+-------+------+-----------+------+-----+---------+-------+---------------------------------+---------+
| Field | Type | Collation | Null | Key | Default | Extra | Privileges                      | Comment |
+-------+------+-----------+------+-----+---------+-------+---------------------------------+---------+
| id    | int  | NULL      | YES  |     | NULL    |       | select,insert,update,references |         |
+-------+------+-----------+------+-----+---------+-------+---------------------------------+---------+
1 row in set (0.05 sec)
```

### 3. What did you see instead (Required)

```sql
mysql> show full columns from t like 'ID';
Empty set (0.00 sec)
```

This leads to incompatibility with some ecosystem tools.

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: 
close #29910 #30803

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by bb7133** [comment]
I don't think it is necessary to introduce 'collation' for `SHOW FULL COLUMNS`, as you can see a some other test cases rather than `SHOW` statements are changed, which is non-trivial(and risky).

Is it easy to make `SHOW FULL COLUMNS` just case-insensitive no matter if new collation is enabled?

**Comment by hawkingrei** [comment]
> I don't think it is necessary to introduce 'collation' for `SHOW FULL COLUMNS`, as you can see a some other test cases rather than `SHOW` statements are changed, which is non-trivial(and risky).
> 
> 
> 
> Is it easy to make `SHOW FULL COLUMNS` just case-insensitive no matter if new collation is enabled?

Have refactored. PTAL.

**Comment by hawkingrei** [comment]
@Defined2014 @djshow832 @tangenta  PTAL

**Comment by hawkingrei** [comment]
> @hawkingrei I've tested your patch against the case in #30357, marked as duplicate, seems still not work yet.

Fixed, Please try against.

**Comment by hawkingrei** [comment]
> @hawkingrei I've tested your patch against the case in #30357, marked as duplicate, seems still not work yet.

FIxed, Please try against

**Comment by hawkingrei** [comment]
> I don't think it is necessary to introduce 'collation' for `SHOW FULL COLUMNS`, as you can see a some other test cases rather than `SHOW` statements are changed, which is non-trivial(and risky).
> 
> Is it easy to make `SHOW FULL COLUMNS` just case-insensitive no matter if new collation is enabled?

Have refactored. PTAL

**Comment by hawkingrei** [comment]
@Defined2014 @djshow832 @tangenta PTAL

File: executor/show_test.go [hunk 1]
```diff
@@ -1674,3 +1674,31 @@ func (s *testSuite5) TestShowTemporaryTable(c *C) {
...
+	tk.MustExec("create table t(id int);")
+	tk.MustExec("create table 中文t(中文id int);")
+	defer tk.Exec("drop table t, 中文t;")
+	// Test for https://github.com/pingcap/tidb/issues/29910
+	tk.MustQuery("show full columns from t like 'ID';").Check(testkit.Rows("id int(11) <nil> YES  <nil>  select,insert,update,references "))
+	tk.MustQuery("show full columns from t like 'id';").Check(testkit.Rows("id int(11) <nil> YES  <nil>  select,insert,update,references "))
+	tk.MustQuery("show full columns from 中文t like '中文ID';").Check(testkit.Rows("中文id int(11) <nil> YES  <nil>  select,insert,update,references "))
+
+	// Test for https://github.com/pingcap/tidb/issues/26253
+	tk.MustQuery("show status like 'ssl_cipher';").Check(testkit.Rows("Ssl_cipher "))
```
Comment (xiongjiwei) [inline executor/show_test.go:1691]: what if `show status like 'ssl_ciphe%';`

Comment (xiongjiwei) [inline executor/show_test.go:1691]: what if `show status like 'ssl_ciphe%';`

Comment (hawkingrei) [reply executor/show_test.go:1691]: if `collation` is not introduced, It is difficult for regexp to ignore case insensitive.

**Comment by zhangyangyu** [comment]
> > @hawkingrei I've tested your patch against the case in #30357, marked as duplicate, seems still not work yet.
> 
> FIxed, Please try against

My case could run as expected. Nice work!

**Comment by wjhuang2016** [comment]
We should refer to CLUSTER_INFO table, use predicate push down to memtable to implement it.
image

**Pull request closed without merging**